File: csharp/src/AutomationFramework.Core/Pages/AccountPage.cs
using AutomationFramework.Core.Support;
using Microsoft.Playwright;

namespace AutomationFramework.Core.Pages;

/// <summary>
/// The signed-in account overview. Reaching this page is the observable outcome of a successful login.
/// </summary>
public sealed class AccountPage
{
    private const string Path = "/account";

    private readonly IPage _page;

    public AccountPage(IPage page)
    {
        _page = page;
        PageTitle = page.GetByTestId("page-title");
        UserMenu = page.GetByTestId("nav-menu");
        SignOutLink = page.GetByTestId("nav-sign-out");
        FavoritesLink = page.GetByTestId("nav-my-favorites");
        ProfileLink = page.GetByTestId("nav-my-profile");
        InvoicesLink = page.GetByTestId("nav-my-invoices");
    }

    public ILocator PageTitle { get; }

    public ILocator UserMenu { get; }

    public ILocator SignOutLink { get; }

    public ILocator FavoritesLink { get; }

    public ILocator ProfileLink { get; }

    public ILocator InvoicesLink { get; }

    public async Task<AccountPage> OpenAsync()
    {
        await Navigation.ToAsync(_page, Path).ConfigureAwait(false);
        return this;
    }

    /// <summary>Opens the user dropdown, which holds sign-out and the account links.</summary>
    public async Task<AccountPage> OpenUserMenuAsync()
    {
        await UserMenu.ClickAsync().ConfigureAwait(false);
        return this;
    }

    public async Task<HomePage> SignOutAsync()
    {
        await OpenUserMenuAsync().ConfigureAwait(false);
        await SignOutLink.ClickAsync().ConfigureAwait(false);
        return new HomePage(_page);
    }
}

File: java/src/main/java/com/company/automation/pages/AccountPage.java
package com.company.automation.pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;

/**
 * The signed-in account overview. Reaching this page is the observable outcome of a successful
 * login.
 */
public final class AccountPage {

  private static final String PATH = "/account";

  private final Page page;

  public final Locator pageTitle;
  public final Locator userMenu;
  public final Locator signOutLink;
  public final Locator favoritesLink;
  public final Locator profileLink;
  public final Locator invoicesLink;

  public AccountPage(Page page) {
    this.page = page;
    this.pageTitle = page.getByTestId("page-title");
    this.userMenu = page.getByTestId("nav-menu");
    this.signOutLink = page.getByTestId("nav-sign-out");
    this.favoritesLink = page.getByTestId("nav-my-favorites");
    this.profileLink = page.getByTestId("nav-my-profile");
    this.invoicesLink = page.getByTestId("nav-my-invoices");
  }

  public AccountPage open() {
    page.navigate(PATH);
    return this;
  }

  /** Opens the user dropdown, which holds sign-out and the account links. */
  public AccountPage openUserMenu() {
    userMenu.click();
    return this;
  }

  public HomePage signOut() {
    openUserMenu();
    signOutLink.click();
    return new HomePage(page);
  }
}
